framework/PostProcessors/EconomicRatio.py

```diff
@@ -0,0 +1,830 @@
+    """
+      Function to handle the parsed paramInput for this class.
+      @ In, paramInput, ParameterInput, the already parsed input.
+      @ Out, None
+    """
+    self.toDo = {}
+    for child in paramInput.subparts:
+      tag = child.getName()
+      if tag in self.scalarVals + self.tealVals:
+        if 'prefix' not in child.parameterValues:
+          self.raiseAnError(IOError, "No prefix is provided for node: ", tag)
+        #get the prefix
+        prefix = child.parameterValues['prefix']
+
+      if tag in self.tealVals:
+        if tag in ['sortinoRatio', 'gainLossRatio']:
+          #get targets
+          targets = set(child.value)
+          if tag not in self.toDo.keys():
+            self.toDo[tag] = [] # list of {'targets':(), 'prefix':str, 'threshold':str}
+          if 'threshold' not in child.parameterValues:
+            threshold = 'zero'
+          else:
+            threshold = child.parameterValues['threshold'].lower()
+            if threshold not in ['zero','median']:
+              self.raiseAWarning('Unrecognized threshold in {}, prefix \'{}\' use zero instead!'.format(tag, prefix))
+              threshold = 'zero'
+          self.toDo[tag].append({'targets':set(targets),
+                                'prefix':prefix,
+                                'threshold':threshold})
+        elif tag in ['expectedShortfall', 'valueAtRisk']:
+          #get targets
+          targets = set(child.value)
+          if tag not in self.toDo.keys():
+            self.toDo[tag] = [] # list of {'targets':(), 'prefix':str, 'threshold':str}
+          if 'threshold' not in child.parameterValues:
+            threshold = 0.05
+          else:
+            threshold = child.parameterValues['threshold']
+            if threshold >1 or threshold <0:
+              self.raiseAnError('Threshold in {}, prefix \'{}\' out of range, please use a float in range (0, 1)!'.format(tag, prefix))
+
+          self.toDo[tag].append({'targets':set(targets),
+                                'prefix':prefix,
+                                'threshold':threshold})
+        else:
+          if tag not in self.toDo.keys():
+            self.toDo[tag] = [] # list of {'targets':(), 'prefix':str}
+          self.toDo[tag].append({'targets':set(child.value),
+                               'prefix':prefix})
+      elif tag in self.scalarVals:
+        if tag not in self.toDo.keys():
+          self.toDo[tag] = [] # list of {'targets':(), 'prefix':str}
+        self.toDo[tag].append({'targets':set(child.value),
+                               'prefix':prefix})
+      elif tag == "pivotParameter":
+        self.pivotParameter = child.value
+
+      else:
+        self.raiseAWarning('Unrecognized node in EconomicRatio "',tag,'" has been ignored!')
+
+    assert (len(self.toDo)>0), self.raiseAnError(IOError, 'EconomicRatio needs parameters to work on! Please check input for PP: ' + self.name)
+
+  def __computePower(self, p, dataset):
```

Comment:
__computePower, __computeVp, __computeEquivalentSampleSize, __computeUnbiasedCorrection, etc. are copy paste from basic statistics...this is extremeley redundant... we need to externalize them from the basic statistics and put them in the MATH  utils.

Reply:
removed  __computeVp, __computeEquivalentSampleSize, __computeUnbiasedCorrection, keeps __computePower might move __computePower to Math uitils future...